tests/otel/index.test.ts

```diff
@@ -0,0 +1,63 @@
+import { describe, it } from 'node:test';
+import * as assert from 'node:assert';
+import * as yaml from 'yaml';
+import { OtelCollectorConfigBuilder } from '../../src/v2/otel/config';
+
+const aws = { config: { region: 'us-west-2' } };
+const prometheusNamespace = 'test-namespace';
+const prometheusWriteEndpoint = 'https://aps-workspaces.us-west-2.amazonaws.com/workspaces/ws-12345/api/v1/remote_write';
+
+describe('OtelCollectorConfigBuilder', () => {
+  it('should generate correct configuration', () => {
+    const yamlOutput = new OtelCollectorConfigBuilder()
+      .withOTLPReceiver(['http'])
+      .withBatchProcessor()
+      .withMemoryLimiterProcessor()
+      .withAPS(prometheusNamespace, prometheusWriteEndpoint, aws.config.region)
+      .withAWSXRayExporter(aws.config.region)
+      .withDebug()
+      .withHealthCheckExtension()
+      .withPprofExtension()
+      .withMetricsPipeline(
+        ["otlp"],
+        ["memory_limiter", "batch"],
+        ["prometheusremotewrite", "debug"]
+      )
+      .withTracesPipeline(
+        ["otlp"],
+        ["memory_limiter", "batch"],
+        ["awsxray", "debug"]
+      )
+      .build();
```

Comment:
Note to self: this could be performed with .withDefault() method (without debug).